File: ShadowFrameworkLite20/src/shadow/system/data/objects/SFShort.java
package shadow.system.data.objects;

import shadow.system.data.SFCharsetObject;
import shadow.system.data.SFInputStream;
import shadow.system.data.SFOutputStream;

public class SFShort extends SFPrimitiveType  implements SFCharsetObject{

	protected short shortValue;
	
	public SFShort(short shortValue) {
		super();
		this.shortValue = shortValue;
	}

	public short getShortValue() {
		return shortValue;
	}

	public void setShortValue(short shortValue) {
		this.shortValue = shortValue;
	}

	@Override
	public void readFromStream(SFInputStream stream) {
		shortValue=stream.readShort();
	}
	
	@Override
	public void writeOnStream(SFOutputStream stream) {
		stream.writeShort(shortValue);
	}
	

	@Override
	public SFShort clone(){
		return new SFShort(shortValue);
	}
	
	@Override
	public void setStringValue(String value) {
		shortValue=new Short(value);
	}
	
	@Override
	public String toStringValue() {
		return shortValue+"";
	}
}

File: ShadowFrameworkJS/src/shadow/system/data/objects/SFShort.js
//Java to JS on 15/03/2012

function SFShort(shortValue){
	this.shortValue=shortValue;
}

inherit(SFShort,SFPrimitiveType);

SFShort.prototype["getShortValue"]=function(){
			return this.shortValue;
		};

SFShort.prototype["setShortValue"]=function(shortValue){
			this.shortValue = shortValue;
		};

SFShort.prototype["readFromStream"]=function(stream){
			this.shortValue = stream.readShort();
		};

SFShort.prototype["writeOnStream"]=function(stream){
			stream.writeShort(this.shortValue);
		};
		
SFShort.prototype["clone"]=function(){
			return new SFShort(this.shortValue);
		};